projects/ngx-prx-styleguide/src/lib/fancy-form/advanced-confirm.directive.stories.ts

```diff
@@ -0,0 +1,108 @@
+import { storiesOf, moduleMetadata } from '@storybook/angular';
+import { action } from '@storybook/addon-actions';
+import { centered } from '@storybook/addon-centered/angular';
+import { withKnobs, text, boolean, select } from '@storybook/addon-knobs';
+import { FancyFormModule } from './fancy-form.module';
+import { BaseModel, RelatedMap } from '../model/base.model';
+import { HalDoc } from '../hal/doc/haldoc';
+import { Observable, of as ofasobservableOf } from 'rxjs';
+import { ModalModule } from '../modal/modal.module';
+
+// Setup simple model for the stories.
+class ConfirmModel extends BaseModel {
+  public foo: string;
+
+  constructor(parent: HalDoc, demo?: HalDoc, loadRelated = true) {
+    super();
+    this.init(parent, demo, loadRelated);
+  }
+
+  SETABLE = ['foo'];
+
+  encode(): {} { return {}; };
+  decode(): void {
+    this.foo = this.doc['foo'];
+  };
+  key(): string { return 'confirm-model'; };
+  related(): RelatedMap { return {}; };
+  saveNew(data: {}): Observable<HalDoc> { return ofasobservableOf(this.doc); };
+};
+const model = new ConfirmModel(null, new HalDoc({
+  foo: 'bar'
+}, null));
+
+// Module metadata for stories.
+const storiesModuleMetaData = moduleMetadata({
+  imports: [
+    FancyFormModule,
+    ModalModule
+  ],
+  schemas: [],
+  declarations: [],
+  providers: [],
+});
+
+storiesOf('Forms Controls|Directives/Advanced Confirm', module)
+  .addDecorator(withKnobs)
+  .addDecorator(centered)
+  .addDecorator(storiesModuleMetaData)
+  .add(
+    'Usage Details (Knobs)',
+    () => {
+      // For some reason, text knob encodes HTML entities. We need to decode them.
+      const decodeHtml = (label: string, value: string) => {
+        const txt = document.createElement('textarea');
+        txt.innerHTML = text(label, value);
+        return txt.value;
+      };
+      const confirmMessage = decodeHtml('Confirmation Message', 'That\'s a risky move');
```

Comment:
this set up seems mildly annoying

Reply:
Yes, it was. I have this smoothed out in the `feat/139-convert-demos-to-stories` branch. Turns out there was a config setting for knobs that I overlooked that prevented HTML encoding in the first place.